include/assimp/BlobIOSystem.h

```diff
@@ -207,27 +213,32 @@ class BlobIOSystem : public IOSystem {
 public:
     // -------------------------------------------------------------------
     const char *GetMagicFileName() const {
-        return AI_BLOBIO_MAGIC;
+        return baseName.empty() ? AI_BLOBIO_MAGIC : baseName.c_str();
     }
 
     // -------------------------------------------------------------------
     aiExportDataBlob *GetBlobChain() {
+        const auto magicName = std::string(this->GetMagicFileName());
+        const bool hasBaseName = baseName.empty();
```

Comment:
Is this correct?  

`const bool hasBaseName = !baseName.empty();`

I guess this version is more correct. What do you think?

Reply:
Indeed, I messed up this check, sorry for that. It is more appropriate to compare the base name to `AI_BLOBIO_MAGIC`, since `GetMagicFileName` already returns the base name, if it is set. I've fixed this and tested both cases using and obj/mtl file:

- Base name set results in blob names "test.obj"/"test.mtl"
- Base name not set results in ""/"mtl", as before.

Thanks for reviewing. 🙂